app/models/manageiq/providers/embedded_ansible/automation_manager/job.rb

```diff
@@ -7,4 +7,26 @@ def retire_now(requester = nil)
     update_attributes(:retirement_requester => requester)
     finish_retirement
   end
+
+  # Intend to be called by UI to display stdout. Therefore the error message directly returned
+  # instead of raising an exception.
+  def raw_stdout_via_worker(userid = User.current_user, format = 'txt')
+    unless MiqRegion.my_region.role_active?("embedded_ansible")
+      return "Cannot get standard output of this playbook because the embedded Ansible role is not enabled"
+    end
+
+    options = {:userid => userid, :action => 'ansible_stdout'}
+    queue_options = {:class_name  => self.class,
+                     :method_name => 'raw_stdout',
+                     :instance_id => id,
+                     :args        => [format],
+                     :priority    => MiqQueue::HIGH_PRIORITY,
+                     :role        => 'embedded_ansible'}
+    taskid = MiqTask.generic_action_with_callback(options, queue_options)
+    MiqTask.wait_for_taskid(taskid)
```

Comment:
@bzwei @himdel 
This is going to block the UI?, are there other ways to fetch the data later given a task.
If we can return a Task and the UI knows to check for it periodically and display the results once available.

Reply:
Yes, this would block the UI.

We have an `initiate_wait_for_task` in ApplicationController... which causes the browser to poll a the task and call a specific action when done.

But you probably can't use it because that would block the whole summary screen from appearing.

Reply:
As far as I can tell, the only safe way of doing that would be to implement the bit in JS - that is:

* render the summary screen, with the stdout field empty, and initiate that task
* replace the html of that field with a component which knows (as a parameter) which task it's waiting for
* that JS component would keep polling the API for the task to complete
* once done, it would display the result

That way, this can error nicely if the server is not available, but won't block all the other stuff from appearing.
(But note that caching the info on the server, same as everything else, would be much much cleaner from the perspective of UI changes involved.)

I can help with the component :)

Reply:
(So whereas all the other fields come from the server as..

```html
<tr class="no-hover" onclick="" title="">
<td class="label">
MAC Address
</td>
<td>

00:50:56:8c:8b:00
</td>
</tr>
```

this would come more as...

```html
<tr class="no-hover" onclick="" title="">
<td class="label">
Raw Stdout</td>
<td>

<raw-stdout task-id="12345"></raw-stdout>
</td>
</tr>
```

Reply:
@himdel What I have is a replacement of current method. I agree a non-blocking method is better. Will a MiqTask sufficient for your proposed work in JS? Can UI simply call `MiqTask.generic_action_with_callback`? If yes then this PR is not needed at all.
I don't understand the part of caching info on the server. Can you explain with more details?

Lastly, the output in general can be large. I think we want to destroy the MiqTask after UI gets the text.

Reply:
@bzwei Well, I'm not quite comfortable with the idea that the UI has wait for all of..

* create a task
* run that task, potentially on a different server
* talk to the provider, if it exists
* finish that task, saving the data to the db

This can be 3 server round trips even assuming the DB is always local - the UI can't possibly block while waiting for all that to happen.

Not mentioning what happens when the provider does not exist, or decided it didn't want to talk :).

---

So, blocking is out... that leaves:

* caching
* loading asynchronously

---

> I don't understand the part of caching info on the server. Can you explain with more details?

Sure .. what I mean is literally what we do for *every* single other thing on that summary screen :) - load it from the DB.

The textual summaries were never designed for any other kind of data, all the fields there come from the DB and are refreshed during the normal Refresh thing.

So.. what I'm asking is, if we can simply not add this as a DB field somewhere, thus reducing this to a problem that's already solved.

(But I *really* don't know if this makes sense.. probably not if that data goes stale every minute or so.)

---

> Will a MiqTask sufficient for your proposed work in JS? Can UI simply call MiqTask.generic_action_with_callback? If yes then this PR is not needed at all.

If we have a method which returns the task id without blocking or talking to ansible first, that's all we need, yes :).

(In the future, additional requirement would be that it needs to be possible to trigger this via the API, but.. since this is a textual summary screen, no point yet.)

Reply:
> If yes then this PR is not needed at all.

..but I still think the method should live in the model - not the UI.

So I think your PR is needed, if you just remove the wait, and just let the method return the task id.

I'll prep the UI PR so we can test it together :).

Reply:
Also @bzwei, could you update your PR to use the SupportsMixin to mark the job as supporting this?

(The current ui logic is `if @job && @job.try(:raw_stdou)`, which is less than optimal :) .. I would hope for something along the lines if `if @job && @job.supports(:raw_stdout)`.)

Reply:
@himdel we may not need SupportsMixin. You can always assume job.raw_stdout is available as long as job is not nil.

Reply:
Aah, OK, I'll update the PR to do that, thanks :)

(Just to be sure .. you do mean any job, not any ansible job, right? Because .. in the UI, this is a job associated with the service.. so.. really potentially any job at all.)

Reply:
The context is embedded ansible playbook. The job is ansible tower job,  not Miq Job.

Reply:
The point is .. this is a job belonging to a service.

Looking at a random service, I can't know the job will be an embedded ansible playbook job. All I know is, it will be a job.

But if you think using the supports mixin here is wrong, I can add an explicit `@record.type == 'ServiceAnsiblePlaybook'`.

Reply:
This applies to `ServiceAnsiblePlaybook` only, as already in https://github.com/ManageIQ/manageiq-ui-classic/pull/2708/files#diff-ef007b9d34027eeec429c666e2186d8eL37. A random service will not provide a job object.

Reply:
Well, both `ServiceAnsiblePlaybook` and `ServiceAnsibleTower` provide a `job` method.

Disregardign the latter (because you're right, we do check that type already), `ServiceAnsiblePlaybook`'s `job` method looks for any resource of the kind `OrchestrationStack`.

Since `OrchestrationStack#raw_stdout_via_worker` definitely fails, leaving only that `type` check seems very brittle.